packages/next/src/server/app-render.tsx

```diff
@@ -1759,18 +1769,26 @@ export async function renderToHTMLOrFlight(
           injectedCSS: new Set(),
           rootLayoutIncluded: false,
         })
+
         const initialTree = createFlightRouterStateFromLoaderTree(loaderTree)
 
+        const metadata = elementsFromResolvedMetadata(
+          await resolveMetadata(metadataItems)
+        )
```

Comment:
Can be done in a later PR but this should be encapsulated in a `<Metadata />` component so it doesn't block datafetching in AppRouter. they should be able to render concurrently and right now metadata will block all other rendering

Reply:
changed to async component, not sure if that works?